src/main/java/luke/commands/CommandResult.java

```diff
@@ -0,0 +1,38 @@
+package luke.commands;
+
+/**
+ * Represents the result of executing a command.
+ */
+public class CommandResult implements Result {
+    private String result;
+    private boolean isExit;
+
+    /**
+     * Constructs a command result with the specified result.
+     *
+     * @param result The specified result for the command.
+     */
+    public CommandResult(String result) {
+        this(result, false);
+    }
+
+    /**
+     * Constructs a command result with the specified result and whether the command would result in exit.
+     *
+     * @param result The specified result for the command.
+     * @param isExit Whether the command would result in an exit status.
+     */
+    CommandResult(String result, boolean isExit) {
+        this.result = result;
+        this.isExit = isExit;
+    }
+
+    public String getResult() {
+        return this.result;
+    }
+
+    public boolean isExit() {
```

Comment:
Hmm, this would create a cyclic dependency between Command and ExitCommand. Also, this class would only contain the results of the command, not the command type itself... It is used for the GUI to print out the result without knowing what command has been executed :) Thanks for the suggestion!